text/0000-view-change-method.md

```diff
@@ -0,0 +1,69 @@
+- Proposal Name: Improve view/change methods in contracts
+- Start Date: 2019-09-26
+- NEP PR: [nearprotocol/neps#0000](https://github.com/nearprotocol/neps/pull/0000)
+
+# Summary
+[summary]: #summary
+
+Currently the separation between view methods and change methods on the contract level is not very well defined and causes
+quite a bit of confusion among developers. We propose in the NEP to elucidate the difference between view methods
+and change methods and how they should be used. In short, we would like to restrict view methods from accessing certain
+context variables and do not distinguish between view and change methods on the contract level. Developers have the option
+to differentiate between the two in frontend or through near-shell.
+
+# Motivation
+[motivation]: #motivation
+
+From the feedback we received it seems that developers are confused by the results they get from view calls, which are
+mainly caused by the context variables such as `signer_account_id`, `current_account_id`, `attached_deposit` do not make
+sense in a view call. To avoid such confusion and create better developer experience, it is better if those context variables
+are prohibited in view calls.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Among context variables, some do make sense in a view call, such as `block_index`, while the majority does not. Here we
```

Comment:
It is confusing to refer to "context variables" because what we are banning is some functions Bindings APIs. Instead of stating which functions are allowed it is better to state which functions are prohibited. For instance, this document implies that `input()` function is prohibited (because it is a part of the context), but actually we don't want to prohibit it. `current_account_id()` also should not be prohibited (which is again implied to be prohibited by this doc), but `signer_account_id()` should, etc. Please follow https://github.com/nearprotocol/NEPs/blob/master/text/0006-bingings.md and list explicitly what functions are prohibited.

Reply:
Also, some functions are not part of `Context API` but are rather part of `Economics API`, like `attached_deposit()`, so it makes sense to not refer to "Context variables" but instead talk about "Bindings methods" because that's what contract developers are going to be interfacing with.